src/components/books/bookCatagories.js

```diff
@@ -1,10 +1,19 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
 import './book.css';
 
 function Catagories() {
+  const { categories } = useSelector((state) => state.categories);
+  const [searchField, setSearchField] = useState('');
+  const [filterResult, setFilterResult] = useState(categories);
+
+  const filterHandler = () => {
+    setFilterResult(filterResult.map((item) => item.country.includes(searchField)));
+  };
   return (
     <div>
-      <button type="button">check status</button>
+      <input type="text" value={searchField} onChange={(e) => setSearchField(e.target.value)} />
+      <button type="button" onClick={filterHandler}>Filter</button>
     </div>
```

Comment:
- [ ] From a previous project you were asked to add a "Check status" button but here you have added a form component. Kindly re-add the previous button for further instructions.

- [ ] The "Check status" button, when clicked, should dispatch an action to the store that fetches the "Under construction" text and displays it on the page. Kindly do that.